server/app/views/applicant/ProgramIndexView.java

```diff
@@ -367,6 +366,21 @@ private LiTag programCard(
         .with(actionDiv);
   }
 
+  /**
+   * If eligibility is gating, the eligibility tag should always show when present. If eligibility
+   * is non-gating, the eligibility tag should only show if the user may be eligible.
+   */
+  private boolean shouldShowEligibilityTag(
+      Http.Request request, ApplicantService.ApplicantProgramData cardData) {
+    if (!featureFlags.isProgramEligibilityConditionsEnabled(request)
+        || !cardData.isProgramMaybeEligible().isPresent()) {
+      return false;
+    }
+    return !featureFlags.isNongatedEligibilityEnabled(request)
```

Comment:
making sure I'm understanding:
Regardless of the value of `isNongatedEligibilityEnabled()`, we want to show the tag if the program has gated eligibility, even if they're not eligible?

Reply:
Yes, when eligibility is gating, we always show all tags.